### PR TITLE
BackupRetention: Fix default selected option to prioritize customer-defined retention

### PR DESCRIPTION
**Pull request by Initsogar**
This PR aims to fix the default retention option to prioritize selecting customer-defined retention by default.

To reproduce the issue, you could access the settings page in Jetpack Cloud with any site with a storage limit (like Jetpack Backup 10 GB) and it will check the retention option according to their plan, and not the one they updated previously.

The expected behavior is to prioritize selecting the customer-defined retention.

| Before | After (expected) |
|---|---|
| image | image |

## Proposed Changes

* Fix to ensure we are deciding whether the default option is the one the customer defined previously or the plan retention period by default.

## Testing Instructions

* Start a Jetpack Cloud live branch
* Navigate to Settings and wait for the new setting to load.
* The Update settings button will be disabled when the current plan or an option that requires an upgrade are selected.
* You could try to update the days of backups saved. The way to re-confirm it was properly saved is refreshing the page.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by jwebbdev — APPROVED** [review]
Tested changing retentions, reloading pages, and navigating to and away from settings to ensure proper retention displays.

**Comment by Initsogar** [comment]
> Tested changing retentions, reloading pages, and navigating to and away from settings to ensure proper retention displays.

@jwebbdev Thank you! It seems there are some issues with the E2E tests and doesn't seem to be related to this change. Either way, I will wait until tomorrow to merge & deploy. Just added the `Ready to merge` label.

**Pull request merged**
